torch/_dynamo/variables/torch.py

```diff
@@ -288,6 +288,9 @@ def call_function(
         ):
             log.warning("Profiler will be ignored")
             return ProfilerContextWrapperVariable(**options)
+        elif self.value is torch.autograd._profiler_enabled:
+            assert not (args or kwargs)
+            return ConstantVariable(torch.autograd._profiler_enabled(), **options)
```

Comment:
This should graph break - you do not want to store the trace time constant into the graph for the sake of control flow. 

Try the following test:

```
 def fn(x):
            if torch.autograd._profiler_enabled():
                with torch.autograd.profiler.record_function("my_function"):
                    return x *  2
            else:
                return x * -1
```
(Note the different result depending on `torch.autograd._profiler_enabled()` value)

1) trace `fn` with `torch.autograd._profiler_enabled()` evaluating to True
2) set `torch.autograd._profiler_enabled()` to False
3) Execute the exported graph

This is an unsafe specialization. Either record this as a call function in the graph, or keep this as a graph break and install a cond operator in its place by hand.

Reply:
I think `torch.autograd._profiler_enabled()` is a global flag indicating whether autograd profiler is enabled so it is fine to store it as a constant?

Reply:
Add the test I described and see.

Reply:
currently we have a few flags traced as constant, https://github.com/pytorch/pytorch/blob/ff6770a9a1db4bb19db24c88bfe7a666722b45d2/torch/_dynamo/config.py#L53-L57 and also a few methods https://github.com/pytorch/pytorch/blob/ff6770a9a1db4bb19db24c88bfe7a666722b45d2/torch/_dynamo/variables/torch.py#L163-L164 as ConstantVariable. In general we don't expect different behavior with profiler enabled or not.

Reply:
graph break and add the test